src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/component/GetRecallComponentTest.kt

```diff
@@ -0,0 +1,91 @@
+package uk.gov.justice.digital.hmpps.managerecallsapi.component
+
+import com.natpryce.hamkrest.Matcher
+import com.natpryce.hamkrest.allOf
+import com.natpryce.hamkrest.assertion.assertThat
+import com.natpryce.hamkrest.equalTo
+import com.natpryce.hamkrest.has
+import org.junit.jupiter.api.Test
+import recallWithPopulatedFields
+import uk.gov.justice.digital.hmpps.managerecallsapi.controller.Api
+import uk.gov.justice.digital.hmpps.managerecallsapi.controller.ApiRecallDocument
+import uk.gov.justice.digital.hmpps.managerecallsapi.controller.ReasonForRecall
+import uk.gov.justice.digital.hmpps.managerecallsapi.controller.RecallResponse
+import uk.gov.justice.digital.hmpps.managerecallsapi.db.Recall
+import uk.gov.justice.digital.hmpps.managerecallsapi.domain.NomsNumber
+import uk.gov.justice.digital.hmpps.managerecallsapi.domain.RecallId
+import uk.gov.justice.digital.hmpps.managerecallsapi.domain.random
+
+class GetRecallComponentTest : ComponentTestBase() {
+
+  private val nomsNumber = NomsNumber("123456")
+
+  @Test
+  fun `get a recall that has just been booked`() {
+    val recallId = ::RecallId.random()
+    recallRepository.save(Recall(recallId, nomsNumber))
+
+    val response = authenticatedGetRequest("/recalls/$recallId")
+
+    assertThat(response, equalTo(RecallResponse(recallId, nomsNumber)))
+  }
+
+  @Test
+  fun `get a fully populated recall`() {
+    val recallId = ::RecallId.random()
+    val fullyPopulatedRecall = recallWithPopulatedFields(recallId, nomsNumber)
+    recallRepository.save(fullyPopulatedRecall)
+
+    val response = authenticatedGetRequest("/recalls/$recallId")
+
+    // MD:  We could alternatively assert on the contents of the json here (and in all endpoint tests) by writing out
+    // the response to a file and asserting that it matches an expected version (obviously would not want to use random
+    // test data for that) That way it's very obvious when a response changes, and we're not serializing and
+    // deserializing. But possibly that sort of assertion should live in the Controller/endpoint specific unit tests.
+    assertThat(
```

Comment:
Yes that was kind of the point of my comment.  Although I think that sort of assertion could live in a lower level test to avoid making this one less readable:  `But possibly that sort of assertion should live in the Controller/endpoint specific unit tests`

Reply:
Changed back to jsonPath assertions for the time being to keep you happy